src/sentry/static/sentry/app/views/asyncView.tsx

```diff
@@ -1,13 +1,10 @@
 import DocumentTitle from 'react-document-title';
 import React from 'react';
 
-import AsyncComponent, {
-  AsyncComponentProps,
-  AsyncComponentState,
-} from 'app/components/asyncComponent';
+import AsyncComponent from 'app/components/asyncComponent';
 
-export type AsyncViewState = AsyncComponentState;
-export type AsyncViewProps = AsyncComponentProps;
+type AsyncViewState = AsyncComponent['state'];
+type AsyncViewProps = AsyncComponent['props'];
```

Comment:
How does typescript know how to map these keys back to the generic?

Reply:
https://stackoverflow.com/a/55005902